iceoryx_posh/test/moduletests/test_roudi_portmanager.cpp

```diff
@@ -439,6 +443,97 @@ TEST_F(PortManager_test, ConditionVariablesDestroy)
     }
 }
 
+TEST_F(PortManager_test, AcquiringMaximumNumberOfEventVariablesWorks)
+{
+    std::string process = "BuddyHolly";
+
+    for (unsigned int i = 0U; i < iox::MAX_NUMBER_OF_EVENT_VARIABLES; i++)
+    {
+        iox::ProcessName_t newProcessName(iox::cxx::TruncateToCapacity, process + std::to_string(i));
+        auto eventVariableDataResult = m_portManager->acquireEventVariableData(newProcessName);
+        EXPECT_THAT(eventVariableDataResult.has_error(), Eq(false));
+    }
+}
+
+TEST_F(PortManager_test, AcquiringOneMoreThanMaximumNumberOfEventVariableFails)
+{
+    std::string process = "BuddyHollysBrille";
+
+    // first acquire all possible event variables
+    for (unsigned int i = 0U; i < iox::MAX_NUMBER_OF_EVENT_VARIABLES; i++)
+    {
+        iox::ProcessName_t newProcessName(iox::cxx::TruncateToCapacity, process + std::to_string(i));
+        auto eventVariableDataResult = m_portManager->acquireEventVariableData(newProcessName);
+        ASSERT_THAT(eventVariableDataResult.has_error(), Eq(false));
+    }
+
+    // test if overflow errors get hit
+    auto errorHandlerCalled{false};
+    auto errorHandlerGuard = iox::ErrorHandler::SetTemporaryErrorHandler(
+        [&errorHandlerCalled](const iox::Error, const std::function<void()>, const iox::ErrorLevel) {
+            errorHandlerCalled = true;
+        });
+
+    auto eventVariableDataResult = m_portManager->acquireEventVariableData("AnotherBrille");
+    EXPECT_THAT(eventVariableDataResult.has_error(), Eq(true));
+    EXPECT_THAT(errorHandlerCalled, Eq(true));
+    EXPECT_THAT(eventVariableDataResult.get_error(), Eq(PortPoolError::EVENT_VARIABLE_LIST_FULL));
+}
+
+TEST_F(PortManager_test, DeletingEventVariableWorks)
+{
+    std::string process = "BudSpencer";
+
+    // first acquire all possible event variables
+    for (unsigned int i = 0U; i < iox::MAX_NUMBER_OF_EVENT_VARIABLES; i++)
+    {
+        iox::ProcessName_t newProcessName(iox::cxx::TruncateToCapacity, process + std::to_string(i));
+        auto eventVariableDataResult = m_portManager->acquireEventVariableData(newProcessName);
+        ASSERT_THAT(eventVariableDataResult.has_error(), Eq(false));
+    }
+
+    // delete one and add one eventVariableDataResult should be possible now
+    unsigned int i = 0U;
+    iox::ProcessName_t newProcessName(iox::cxx::TruncateToCapacity, process + std::to_string(i));
+    m_portManager->deletePortsOfProcess(newProcessName);
+
+    auto eventVariableDataResult = m_portManager->acquireEventVariableData(newProcessName);
+    EXPECT_THAT(eventVariableDataResult.has_error(), Eq(false));
+}
+
+TEST_F(PortManager_test, DestroyEventVariableAndAddNewOneSucceeds)
```

Comment:
This file is a nice proof of the broken window theory. Since it doesn't follow our best practice, new test also apply the bad habit :)